modules/proxy/daicard.io/entry.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Set default email & domain name
+domain="${DOMAINNAME:-localhost}"
+email="${EMAIL:-noreply@gmail.com}"
+indra_url="$INDRA_URL"
+echo "domain=$domain email=$email indra=$indra_url"
+
+# Provide a message indicating that we're still waiting for everything to wake up
+function loading_msg {
+  while true # unix.stackexchange.com/a/37762
+  do echo 'Waiting for the rest of the app to wake up..' | nc -lk -p 80
+  done > /dev/null
+}
+loading_msg &
+loading_pid="$!"
+
+########################################
+# Wait for downstream services to wake up
+# Define service hostnames & ports we depend on
+
+echo "waiting for ${indra_url#*://}..."
+bash wait_for.sh -t 60 ${indra_url#*://} 2> /dev/null
+while ! curl -s $indra_url > /dev/null
+do sleep 2
+done
+
+# Kill the loading message server
+kill "$loading_pid" && pkill nc
+
+########################################
+# Setup SSL Certs
+
+letsencrypt=/etc/letsencrypt/live
+devcerts=$letsencrypt/localhost
+mkdir -p $devcerts
+mkdir -p /etc/certs
+mkdir -p /var/www/letsencrypt
+
+if [[ "$domain" == "localhost" && ! -f "$devcerts/privkey.pem" ]]
+then
+  echo "Developing locally, generating self-signed certs"
+  openssl req -x509 -newkey rsa:4096 -keyout $devcerts/privkey.pem -out $devcerts/fullchain.pem -days 365 -nodes -subj '/CN=localhost'
+fi
+
+if [[ ! -f "$letsencrypt/$domain/privkey.pem" ]]
+then
+  echo "Couldn't find certs for $domain, using certbot to initialize those now.."
+  certbot certonly --standalone -m $email --agree-tos --no-eff-email -d $domain -n
+  [[ $? -eq 0 ]] || sleep 9999 # FREEZE! Don't pester eff & get throttled
+fi
+
+echo "Using certs for $domain"
+ln -sf $letsencrypt/$domain/privkey.pem /etc/certs/privkey.pem
+ln -sf $letsencrypt/$domain/fullchain.pem /etc/certs/fullchain.pem
+
+# Hack way to implement variables in the nginx.conf file
+sed -i 's/$hostname/'"$domain"'/' /etc/nginx/nginx.conf
+sed -i 's|$INDRA_URL|'"$indra_url"'|' /etc/nginx/nginx.conf
+sed -i 's|$RELAY_URL|'"http://${indra_url#*://}:4223"'|' /etc/nginx/nginx.conf
```

Comment:
This doesn't work w https, not sure why.. idk how big of a problem it is to use http here..